src/SSHDebugPS/VSOperationWaiter.cs

```diff
@@ -30,39 +22,21 @@ internal static class VSOperationWaiter
         /// <exception cref="OperationCanceledException">Wait was canceled and 'throwOnCancel' is true</exception>
         public static bool Wait(string actionName, bool throwOnCancel, Action action)
         {
-            Task t = Task.Run(action);
-
-            DebugEngineHost.HostWaitLoop waiterImpl = null;
+            bool success = false;
 
-            try
-            {
-                waiterImpl = new DebugEngineHost.HostWaitLoop(actionName);
-            }
-            catch (FileNotFoundException)
-            {
-                // Visual Studio is not installed on this box
-            }
-
-            if (waiterImpl != null)
-            {
-                using (ManualResetEvent completeEvent = new ManualResetEvent(false))
+            ThreadHelper.JoinableTaskFactory.Run(
+                actionName,
```

Comment:
>actionName [](start = 16, length = 10)

Does this overload of JTF.Run put up a threaded wait dialog if things take too long?

Reply:
Yes. But it also gives a Cancel button on the dialog. So we should either switch to an overload that does not offer the delegate a `CancellationToken` (and doesn't display the Cancel button to the user) or we should make `action` be cancelable.